daemon/create_windows.go

```diff
@@ -2,20 +2,41 @@ package daemon
 
 import (
 	"fmt"
+	"strings"
 
 	containertypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/container"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/volume"
 )
 
+// dedupConfigEnvVars deduplicates requested environment variables in the config
+// as they are case insensitive on Windows
+func dedupConfigEnvVars(config *containertypes.Config) {
+	deduped := make(map[string]string)
+	for _, v := range config.Env {
+		i := strings.Index(v, "=")
+		if i >= 0 {
+			name := strings.ToUpper(v[:i])
+			value := v[i+1:]
+			deduped[name] = value
+		}
+	}
+	config.Env = nil
+	for k, v := range deduped {
+		config.Env = append(config.Env, k+"="+v)
+	}
+}
```

Comment:
This PR has same issues as https://github.com/docker/docker/pull/28725#pullrequestreview-9928652. To keep the current behavior we could only fix it in https://github.com/docker/docker/blob/master/daemon/commit.go#L46-L56 so that it keeps all user envs and skips image ones with a case insensitive compare.